core/src/main/scala/org/apache/spark/util/VersionUtils.scala

```diff
@@ -36,6 +37,19 @@ private[spark] object VersionUtils {
    */
   def minorVersion(sparkVersion: String): Int = majorMinorVersion(sparkVersion)._2
 
+  /**
+   * Given a Spark version string, return the short version string.
+   * E.g., for 3.0.0-SNAPSHOT, return '3.0.0'.
```

Comment:
I didn't change this example in order to minimize the diff between `branch-2.4` and `master`.